ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/storage/keyvalue/RocksDbStorageProvider.java

```diff
@@ -27,13 +27,23 @@
 
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class RocksDbStorageProvider {
   public static long DEFAULT_WORLD_STATE_PREIMAGE_CACHE_SIZE = 5_000L;
   private static final Logger LOG = LogManager.getLogger();
+  public static final int CURRENT_VERSION = 1;
+  /**
+   * This list contains the methods used to load the version at its index. Index 0 of the list loads
+   * a version 0 database and so on...
+   */
+  private static final List<StorageProviderFunction> PROVIDERS_BY_VERSION =
+      Arrays.asList(RocksDbStorageProvider::ofUnsegmented, RocksDbStorageProvider::ofSegmented);
```

Comment:
Would a Map<Integer,StorageProviderFunction> work better in the event that version numbers are non-consecutive?  For example if we drop support for a specific experimental version?

Reply:
Good idea!